contrib/pg_upgrade/info.c

```diff
@@ -649,6 +649,9 @@ get_rel_infos(ClusterInfo *cluster, DbInfo *dbinfo)
 				last_tablespace = curr->tablespace = pg_strdup(tablespace);
 				curr->tblsp_alloc = true;
 			}
+
+			if (tablespace != PQgetvalue(res, relnum, i_spclocation))
```

Comment:
Can you add to description more details why the mem free is conditional and why using of this check is correct?

Reply:
added